### PR TITLE
Implement remaining number formatting functions

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Presented as an alternative to #2118. ~Clearly repeating the same pattern, so can probably use templates to deduplicate a bunch.~

### CODE REVIEW

File: src/OpenLoco/src/Localisation/Formatting.cpp [hunk 1]
```diff
@@ -217,7 +340,7 @@ namespace OpenLoco::StringManager
         const char* prefixSymbol = getString(currency->prefixSymbol);
         buffer = formatStringPart(buffer, prefixSymbol, nullptr);
 
-        buffer = formatInt48Grouped(localisedValue, buffer, currency->separator);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Localisation/Formatting.cpp:220]: Not sure what to do with this. #2118 treats it as `value * static_cast<uint64_t>(std::pow(10, separator))`??

**Comment by duncanspumpkin** [comment]
I'd much prefer we just use libfmt rather than reinvent the wheel.

**Comment by AaronVanGeffen** [comment]
> I'd much prefer we just use libfmt rather than reinvent the wheel.

Indeed, which is what we tried in #2118. I agree that, ideally, we wouldn't reinvent the wheel, but I felt #2118 wasn't going anywhere. We kept running into locale problems, though, and regressions on this terrain are not game breaking imo.

Given recent efforts with no_loco builds, I wanted to present an alternative. Given the scope, I think it's a good compromise.

**Review by ZehMatt — APPROVED** [review]
I think we should have this in for the time being, it does help refactor this area when I don't have to deal with interop code.

**Pull request closed without merging**